### PR TITLE
test: add cypress jest plugin for better expectations

### PR DESCRIPTION
**Pull request by danWatkinson**
## Ticket Number
AS-102

## Description of change
suggested better way to assert against our payload during 'submission to horizon' tests

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: e2e-tests/cypress/integration/appeal-submission-to-horizon.feature [hunk 1]
```diff
@@ -11,8 +10,8 @@ Feature: Appeal submission to Horizon - create case for appellant
     When the appeal is submitted
     Then a case is created for a case officer
 
-  @ac-102-ac1
-  Scenario: Appeal information submitted by an Agent 
-    Given an agent has provided appeal information
-    When the appeal is submitted
-    Then a case is created for a case officer
\ No newline at end of file
+  # @ac-102-ac1
```
Comment (danWatkinson) [inline e2e-tests/cypress/integration/appeal-submission-to-horizon.feature:13]: sorry, this is obviously a bit dirty but i needed to have the tests running in the new format hence removing @wip and commenting out the new scenario
obviously in the actual branch this scenario should be in play..

File: e2e-tests/cypress/integration/appeal-submission-to-horizon/appeal-submission-to-horizon.js [hunk 2]
```diff
@@ -1,6 +1,51 @@
 import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
 const { APPEAL_DOCUMENT } = require('../../../../packages/forms-web-app/src/lib/empty-appeal');
 
+const matchWhatWeCanFrom = (hardCodedExpectations) => {
```
Comment (danWatkinson) [inline e2e-tests/cypress/integration/appeal-submission-to-horizon/appeal-submission-to-horizon.js:4]: medium term this probably pops out into a helper or something, but it's likely good enough to work for the existing scenario + the new one we're adding in this branch..
if it stays in this branch it'll get tidied in the next 'horizon submission' story so up to you how hardcore you want to be here ;)

File: e2e-tests/cypress/support/index.js [hunk 3]
```diff
@@ -15,7 +15,9 @@
 
 // Import commands.js using ES2015 syntax:
 import './commands';
-import "cypress-ntlm-auth/dist/commands";
+import 'cypress-ntlm-auth/dist/commands';
+import 'cypress-jest-adapter';
```
Comment (danWatkinson) [inline e2e-tests/cypress/support/index.js:19]: the thing that lets us do the old expect.any(String) type matching

**Review by Fenisay — APPROVED** [review]
LGTM

**Pull request merged**